addons/stock/models/stock_move.py

```diff
@@ -1237,7 +1237,7 @@ def _search_picking_for_assignation_domain(self):
             ('picking_type_id', '=', self.picking_type_id.id),
             ('printed', '=', False),
             ('state', 'in', ['draft', 'confirmed', 'waiting', 'partially_available', 'assigned'])]
-        if self.partner_id and (self.location_id.usage == 'transit' or self.location_dest_id.usage == 'transit'):
+        if self.partner_id:
```

Comment:
if this condition is too lax 
try self.partner_id any(loc.usage in ( 'transit', 'supplier', 'customer') for loc in (self.location_id, self.location_dest_id)):